### DISCUSSION TITLE
useSwitchNetwork unable to add network

### DISCUSSION THREAD

**Discussion by SaadNiaziDev** [wevm/wagmi#1965] [Q&A]
Currently having issue to switcjing to chain which is not added in my metamask android app.
I receive following error

``
SwitchChainError: Error switching chain
    at WalletConnectLegacyConnector.switchChain_fn [as switchChain] (walletConnectLegacy.js:167:11)
``

useSwitchNetwork hook will work flawlessly if network is already added but wont if not there. It use to work i guess but something happened and its not working

wagmi :0.12.0

**Comment by paulclindo** [comment]
I'm having the same issue whenever I want to switch network and the network is not added on metamask mobile app and it seems that on  `switchNetwork` from [InjectedConnector](https://github.com/wagmi-dev/references/blob/9cb0535077504c27626b2e4bc32dc983d63a56ba/packages/connectors/src/injected.ts#L230-L236) (If I'm not mistaken) is only handling the error code `4902` and for metamask mobile we have a different code `-32603` . See more on this issue https://github.com/MetaMask/metamask-mobile/issues/2944#issuecomment-976988719

> Metamask on mobile does not throw error.code = 4902 for some reason. error.code is -32603.
